### PR TITLE
[clang][Interp] Handle std::move etc. builtins

### PR DESCRIPTION
**Pull request by tbaederr**
Only light testing here since I'm not sure how to properly test this.

### CODE REVIEW

**Comment by hnrklssn** [comment]
Would making a class with custom move and copy constructors with side effects, and moving/copying an instance around a bit to affect the results of static_asserts, work for testing?

**Comment by hnrklssn** [comment]
Nice. I realise you're following the convention already established in the test file, but have you considered compiling with `-verify=new,both` and `-verify=ref,both`, respectively, and combining the shared diagnostics into `both-error` etc. to make it easier to see when they do and don't align?

**Comment by tbaederr** [comment]
> Nice. I realise you're following the convention already established in the test file, but have you considered compiling with `-verify=new,both` and `-verify=ref,both`, respectively, and combining the shared diagnostics into `both-error` etc. to make it easier to see when they do and don't align?

That might be an interesting approach for the next new test file I add, thanks.

**Comment by tbaederr** [comment]
Ping

**Comment by tbaederr** [comment]
Ping

**Comment by hnrklssn** [comment]
LGTM, but someone else should approve also.

**Comment by tbaederr** [comment]
Ping

File: clang/test/AST/Interp/functions.cpp
Comment (AaronBallman) [inline clang/test/AST/Interp/functions.cpp:392]: I'd like to see test coverage involving a move constructor, a move assignment operator, a direct call to __builtin_move, and some testing for `std::as_const` and `std::forward`. Of special interest would be times when there's UB in the move constructor/move assignment that should be caught or a non-copyable object where move semantics are the only thing that should work.

**Review by AaronBallman — APPROVED** [review]
LGTM assuming no surprises come up from the additional test coverage.

File: clang/test/AST/Interp/functions.cpp
Comment (tbaederr) [reply clang/test/AST/Interp/functions.cpp:427]: I've enabled `test/SemaCXX/builtin-std-move.cpp` with the new interpreter.

**Pull request merged**
